File: tools/docs-generator/lib/physics/shape/vertex_prefab_xml_reader.py
import xml.etree.ElementTree as EXml

from lib.physics.shape.vertex_prefab import VertexPrefab


class VertexPrefabXmlReader(object):
    VERTICES_ELEMENT = "vertices"
    VERTEX_ELEMENT = "vertex"
    X_ATTRIBUTE = "x"
    Y_ATTRIBUTE = "y"

    def read_prefabs_from_string(self, xml_string):
        vertices = []
        element = EXml.fromstring(xml_string)
        for vertex in element.findall(self.VERTEX_ELEMENT):
            vertices.append(
                self.read_prefab_from_string(EXml.tostring(vertex))
            )
        return vertices

    def read_prefab_from_string(self, xml_string):
        try:
            element = EXml.fromstring(xml_string)
            return VertexPrefab(
                float(element.get(self.X_ATTRIBUTE)),
                float(element.get(self.Y_ATTRIBUTE))
            )
        except Exception as e:
            raise VertexPrefabXmlReadException(xml_string, e)


class VertexPrefabXmlReadException(Exception):
    MESSAGE_TEMPLATE = "Cannot read vertex prefab from xml string {}. Cause: {}."

    def __init__(self, xml_string, cause):
        super().__init__(self.MESSAGE_TEMPLATE.format(xml_string, cause))

File: core/src/remove/tanks/game/physics/shape/VertexPrefabXmlReader.java
package remove.tanks.game.physics.shape;

import com.badlogic.gdx.utils.XmlReader;

import java.util.Arrays;
import java.util.List;
import java.util.stream.Collectors;

/**
 * @author Mateusz Długosz
 */
public final class VertexPrefabXmlReader {
    public static final String VERTEX_ELEMENT = "vertex";
    public static final String VERTICES_ELEMENT = "vertices";

    private static final String X_ATTRIBUTE = "x";
    private static final String Y_ATTRIBUTE = "y";

    public List<VertexPrefab> readVerticesPrefabs(XmlReader.Element element) {
        return Arrays.stream(element.getChildrenByName(VERTEX_ELEMENT).toArray())
                .map(this::readVertexPrefab)
                .collect(Collectors.toList());
    }

    public VertexPrefab readVertexPrefab(XmlReader.Element element) {
        try {
            return new VertexPrefab(
                    element.getFloat(X_ATTRIBUTE),
                    element.getFloat(Y_ATTRIBUTE)
            );
        } catch (Exception e) {
            throw new VertexPrefabXmlReadException(element, e);
        }
    }
}
